### ISSUE TITLE
package.json not found error is misleading

### ISSUE THREAD

**Issue by dmfenton** [dawson-org/dawson-cli#158]
```
➜  koop-provider-craigslist git:(master) ✗ dawson dev

   Execution Error
   Cannot find package.json

   Possible cause
   There is no package.json file in the current directory

   You are running this command from '/Users/foobar/repos/koop-provider-craigslist' which does not
   contain a package.json file as required by dawson.

   Possible solution
   * check if the file exists by running 'stat /Users/foobar/repos/koop-provider-craigslist/package.json'
   * run dawson from the correct folder
   * check file permissions on package.json

   If you believe this is not an expected behaviour or that this error message should be improved, open an issue: https://github.com/dawson-org/dawson-cli/issues
➜  koop-provider-craigslist git:(master) ✗ stat /Users/foobar/repos/koop-provider-craigslist/package.json
16777220 167870324 -rwxrwxrwx 1 dani7825 (1968095660 0 1005 "Mar 25 19:14:26 2017" "Mar 25 19:12:55 2017" "Mar 25 19:14:17 2017" "Mar 25 19:12:55 2017" 4096 8 0 /Users/foobar/repos/koop-provider-craigslist/package.json
```

Package.json exists but there is a syntax error

https://github.com/dmfenton/koop-provider-craigslist/tree/dawson

**Comment by alexcasalboni** [comment]
Hi @dmfenton,

if you check [your file on Github](https://github.com/dmfenton/koop-provider-craigslist/blob/dawson/package.json) you can already spot the syntax error :)

@lusentis I guess the error message could be improved?

**Comment by dmfenton** [comment]
Thanks! I updated this issue to reflect the confusing error message.

**Comment by dmfenton** [comment]
I fixed the syntax error but now I get this

```
dawson dev
   Execution Error
   dawson configuration error

   Possible cause
   Path specified by 'assetsDir' does not exist.

   Possible solution
   Directory does not exist: '/Users/foobar/repos/koop-provider-craigslist/assets',
   either create this directory, set the correct value for the 'assetsDir' property
   in package.json, or set 'assetsDir' to false if you're not using static assets.
```

**Comment by alexcasalboni** [comment]
@dmfenton quoting the [documentation](https://dawson.sh/docs.html#-12): "_Specify false to skip deploying assets_", and I assume that's what you are trying to do.

Looking at [the code](https://github.com/dawson-org/dawson-cli/blob/675a49095ae378d729b248bee955fc2970bd3133/src/config.js#L171), it seems like such error message would never show up if `assetsDir` is actually `false`.

In case you are not specifying `false`, then could you share your current configuration? (your [package.json](https://github.com/dmfenton/koop-provider-craigslist/blob/dawson/package.json) doesn't seem to be up to date?)

**Comment by dmfenton** [comment]
assetsDir is false and this is showing up. I just pushed up the code again

**Comment by alexcasalboni** [comment]
@dmfenton I think the problem is that `assetsDir` should be defined inside a `dawson` property in your package.json file, as documented [here](https://dawson.sh/docs.html#5-application-configuration).

**Comment by dmfenton** [comment]
yep, that did it. thank you. I must say I'm not sure I understood what Dawson Dev was going to do. I thought it was going to start a server locally, but looks like its actually pushing things to AWS?

**Comment by alexcasalboni** [comment]
You are welcome.

Yes, Dawson will run a local server if you execute `dawson dev`. The [local development documentation](https://dawson.sh/docs.html#7-working-with-the-development-server) is not complete yet, but it will be very soon :)

**Comment by lusentis** [comment]
Thanks to everyone,

- we'll add a check so we can determine whether the `package.json` contains a syntax error.
- we'll relax the check on `assetsDir` so that a missing directory won't cause the command to fail.

> I thought it was going to start a server locally, but looks like its actually pushing things to AWS?

dev server docs are coming asap, in the meantime (any suggestion / PR to make this more clear will be really appreciated):

`dawson dev`:
- simulates (*some parts of*) Amazon CloudFront + Amazon API Gateway + AWS Lambda locally
- allows you to execute your function's code without needing to deploy to AWS every time you make a change
- functions will be executed using *real* `AWS IAM Role`. Such IAM Roles need to exist on AWS, this is why dawson dev will ask you to run `dawson deploy` first (no other resource is currently used internally by the dev server)
- additionally, any other resource such as S3 Buckets, DynamoDB Tables, EC2 Instances etc (if defined by you) cannot be emulated locally - so they are required to exist on AWS

@dmfenton Hope this clarifies your expectations, if not, please let us know!

Thanks for your feedback,

**Comment by lusentis** [comment]
@dmfenton It would be awesome if you could start a PR to fix this message.
Starting from this line: https://github.com/dawson-org/dawson-cli/blob/master/src/config.js#L354
I'd do an `if` to check if the error message matches "SyntaxError" and display the appropriate message.

**Comment by dmfenton** [comment]
I appreciate your help on this, but I've decided to go with Claudia.js for now. Will check back in with this project sometime in the future!

**Comment by lusentis** [comment]
fixed in 828381c9efc75a6b01cfcd132daaf34d76ff09af, will be released in v0.28 soon.

**Comment by lusentis** [comment]
also, 56156cab59b8581a08220c412c1a007a70735043 fixes that `assetsDir` behaviour and allows deploying without having to specify assetsDir = false.
